### ISSUE TITLE
Broken existing ExternalSecrets because of empty decodeStrategy.

### ISSUE THREAD

**Issue by snarlysodboxer** [external-secrets/external-secrets#1373]
**Describe the solution you'd like**
Fix empty decode strategy bug.

**What is the added value?**
It should be allowed to not set decodeStrategy.

**Give us examples of the outcome**

Currently, since https://github.com/external-secrets/external-secrets/pull/1294, if you do not specify a decodeStrategy on your ExternalSecret specs, you get an error like `"error":"could not apply decoding strategy to spec.data[0]: decoding strategy  is not supported"` (notice the double spaces at `strategy  is not supported` where [this line](https://github.com/external-secrets/external-secrets/blob/main/pkg/utils/utils.go#L83) inserts an empty string.)

In other words, it seems the code is not properly defaulting to the `None` strategy.

Testing with an ExternalSecret spec like this:
```yaml
  data:
  - secretKey: .dockerconfigjson
    remoteRef:
      key: docker-credentials
      property: docker-credentials.json
```

**Comment by snarlysodboxer** [comment]
My mistake! I was testing against the previous versions of the CRDs. - Once I applied the updated CRDs, the `decodingStrategy` now defaults to `None`, which fixes the issue. Sorry for the false alarm.
